src/services/stringCompletions.ts

```diff
@@ -838,17 +838,20 @@ function getCompletionEntriesForNonRelativeModules(
     extensionOptions: ExtensionOptions,
     typeChecker: TypeChecker,
 ): readonly NameAndKind[] {
-    const { baseUrl, paths } = compilerOptions;
+    const { baseUrl, paths, pathsBasePath } = compilerOptions;
 
     const result = createNameAndKindSet();
     const moduleResolution = getEmitModuleResolutionKind(compilerOptions);
+
+    const projectDir = compilerOptions.project || host.getCurrentDirectory();
+
+    const absolute = normalizePath(combinePaths(projectDir, baseUrl ?? pathsBasePath));

```

Comment:
Take a look at the utility function `getPathsBasePath`. It takes `baseUrl` and the host’s current directory into account as needed. It should return an absolute directory as long as `paths` is defined.

Reply:
Do you mean I should use `getPathsBasePath` in paths entries and leave the baseUrl entries using the original logic? Or the absolute variable can be `getNormalizedAbsolutePath(getPathsBasePath(compilerOptions, host)!, projectDir)`.? 

I don't know if there is a situation where there is a difference. Also not sure if `getNormalizedAbsolutePath` and `projectDir` is necessary. in both VSCode and fourslash test, the `baseUrl` here is absolute and  `compilerOptions.project` is always undefined.

Reply:
Yeah, I think what you have is correct, but you should use the existing `getPathsBasePath` to make this code a bit more concise and ensure the logic for making the absolute path is reused in a single place.